android/src/test/java/com/rudderstack/android/plugins/ExtractStatePluginTest.kt

```diff
@@ -14,97 +14,183 @@
-
-        //TODO(add tests. currently we are not using core)
-//        assertThat(storage.traits, allOf(aMapWithSize(1),
-//        hasEntry("userId", "userId")))
-//        assertThat(storage.externalIds, iterableWithSize(1))
-//        assertThat(
-//            storage.externalIds!![0], allOf(aMapWithSize(1),
-//            hasEntry("dest-1", "dest-1-id")))
+    fun `intercept should proceed with modified message for IdentifyMessage with anonymousId`() {
+        val identifyMessage = IdentifyMessage.create(traits = mapOf("anonymousId" to null, "userId" to "userId"), timestamp = RudderUtils.timeStamp)
+        `when`(chain.message()).thenReturn(identifyMessage)
+
+        plugin.intercept(chain)
+        val messageCaptor = argumentCaptor<IdentifyMessage>()
+        verify(chain).proceed(messageCaptor.capture())
+        println("118,${messageCaptor.allValues}")
+        val capturedMessage = messageCaptor.lastValue
+        MatcherAssert.assertThat(capturedMessage.context?.traits?.get("anonymousId"), Matchers.notNullValue())
     }
+
     @Test
-    fun `test identify with both options and message external ids`(){
-        val storage = BasicStorageImpl( )
-        val options = RudderOptions.Builder().withExternalIds(listOf(mapOf(
-            "dest-1" to "dest-1-id"
-        ),mapOf(
-            "dest-2" to "dest-2-id"
-        )
-        )).build()
-        val extractStatePlugin = ExtractStatePlugin()
-        val identifyMsg = IdentifyMessage.create(timestamp = RudderUtils.timeStamp,
-        traits = mapOf("userId" to "userId"), externalIds = listOf(mapOf(
-                "dest-3" to "dest-3-id"
-            ), mapOf(
-                "dest-2" to "not-dest-2-id"
-            ) ))
-//        val centralPluginChain = CentralPluginChain(identifyMsg, listOf(RudderOptionPlugin(options),
-//            extractStatePlugin))
-//        centralPluginChain.proceed(identifyMsg)
-//        assertThat(storage.traits, allOf(aMapWithSize(1),
-//        hasEntry("userId", "userId")))
-//        assertThat(storage.externalIds, allOf(iterableWithSize(3),
-//        everyItem(
-//            aMapWithSize(1)
-//        ), containsInAnyOrder(
-//                mapOf(
-//                    "dest-1" to "dest-1-id"
-//                ),mapOf(
-//                    "dest-2" to "not-dest-2-id"
-//                ),
-//                mapOf(
-//                    "dest-3" to "dest-3-id"
-//                )
-//        )
-//        ))
+    fun `intercept should update context with new userId for AliasMessage`() {
+        val aliasMessage = AliasMessage.create(RudderUtils.timeStamp, userId = "newUserId")
+        `when`(chain.message()).thenReturn(aliasMessage)
+        plugin.intercept(chain)
+        val messageCaptor = argumentCaptor<AliasMessage>()
+
+        verify(chain).proceed(messageCaptor.capture())
+        val capturedMessage = messageCaptor.lastValue
+        assertEquals("newUserId", capturedMessage.userId)
+    }
+
+ /*   @Test
```

Comment:
If all the commented-out tests are not needed, can you please consider removing them?

Reply:
No, we need to fix them later for better test coverage. Will get back to this when other PRs are merged.